Image checker issues

For unit tests image checking, error messages as follows:
Updated:
```
Traceback (most recent call last):
File "/home/runner/work/e3sm_diags/e3sm_diags/tests/system/test_diags.py", line 336, in test_image_count
self.assertEqual(actual_num_images, expected_num_images)
AssertionError: 30 != 28
```

For complete run checking, the viewer output directory structure changed with `output_viewer` 1.3.3 update. The checker needs to be adjusted as well. ex. `area_mean_time_series/TREFHT.png` is now located at `area_mean_time_series/TREFHT/TREFHT.png. 
```
======================================================================
ERROR: test_complete_run (tests.complete_run.TestCompleteRun)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.zhang40/e3sm_diags/tests/complete_run.py", line 28, in test_complete_run
    compare_images(
  File "/gpfs/fs1/home/ac.zhang40/e3sm_diags/tests/system/test_diags.py", line 75, in compare_images
    actual_png = Image.open(path_to_actual_png).convert("RGB")
  File "/home/ac.zhang40/y/envs/ed250/lib/python3.8/site-packages/PIL/Image.py", line 2912, in open
    fp = builtins.open(filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/lcrc/group/e3sm/public_html/diagnostic_output/zhang40/complete_runs/2_5_0_try1/v2_4_0_all_sets/area_mean_time_series/TREFHT.png'
 
----------------------------------------------------------------------
Ran 1 test in 860.019s
 
FAILED (errors=1)
``` 
We can perhaps turn off image checker for `rc2` and bring it back when this is fixed for next `rc` or final release.

--- a/conda/e3sm_diags_env.yml
+++ b/conda/e3sm_diags_env.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
   # Base
   # ==============
-  - python>=3.9
+  - python=3.9.4
   - pip=21.1.2
   - numpy=1.20.3
   - matplotlib=3.4.2

--- a/conda/e3sm_diags_env_dev.yml
+++ b/conda/e3sm_diags_env_dev.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
   # Base
   # =================
-  - python>=3.9
+  - python=3.9.4
   - pip=21.1.2
   - numpy=1.20.3
   - matplotlib=3.4.2

--- a/tests/download_data.py
+++ b/tests/download_data.py
@@ -74,7 +74,7 @@ def download():
     )
     print("Downloading unit_test_images")
     download_files(
-        "https://web.lcrc.anl.gov/public/e3sm/e3sm_diags_test_data",
+        "https://web.lcrc.anl.gov/public/e3sm/e3sm_diags_test_data/system_test/expected",
         "unit_test_images",
     )
     print("Downloaded unit_test_data and unit_test_images")

--- a/tests/system/test_diags.py
+++ b/tests/system/test_diags.py
@@ -56,11 +56,14 @@ def move_to_web(machine_path_re_str, html_prefix_format_str, results_dir):
 
 def count_images(directory):
     images = []
-    for _, _, filenames in os.walk(directory):
-        for file in filenames:
-            if file.endswith(".png"):
-                images.append(file)
-    return len(images)
+    for root, _, filenames in os.walk(directory):
+        # download_data.py won't download files in the viewer directory
+        # because the webpage is more than a simple page of links.
+        if "viewer" not in root:
+            for file in filenames:
+                if file.endswith(".png"):
+                    images.append(file)
+    return len(images), images
 
 
 def compare_images(
@@ -331,8 +334,9 @@ class TestAllSets(unittest.TestCase):
 
     # Test the image count
     def test_image_count(self):
-        actual_num_images = count_images("all_sets_results_test")
-        expected_num_images = count_images("../unit_test_images")
+        actual_num_images, actual_images = count_images("all_sets_results_test")
+        expected_num_images, expected_images = count_images("../unit_test_images")
+        self.assertEqual(actual_images, expected_images)
         self.assertEqual(actual_num_images, expected_num_images)
 
     # Test sets

